core/src/SYCL/Kokkos_SYCL_Instance.cpp

```diff
@@ -193,6 +196,10 @@ void SYCLInternal::finalize() {
                       m_instance_id);
   was_finalized = true;
 
+  // The global_unique_token_locks array is static and should only be
+  // deallocated once by the defualt instance
+  if (this == &singleton()) Impl::sycl_global_unique_token_locks(true);
```

Comment:
No warning for discarding the return value?

Reply:
I didn't see one so I don't see much of a reason to cast it.